router.js

```diff
@@ -12,20 +12,27 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
   Basic router that calls a callback whenever the location is updated.
 
   Sample use:
-  
+
+    import { installRouter } from '../node_modules/pwa-helpers/router.js';
+    //...
+    installRouter((location) => _locationChanged(location));
+
+  For example, if you're using this router in a Redux-connected component,
+  you could dispatch an action in the callback:
+
     import { installRouter } from '../node_modules/pwa-helpers/router.js';
     import { navigate } from '../actions/app.js';
+    //...
+    installRouter((location) => store.dispatch(navigate(location)))
+
 
-  If you don’t have any other work to do other than dispatching an action,
-  you can write something like:
+  If you need to force a navigation to a new location programmatically, you can
+  do so by pushing a new state using the History API:
 
-    installRouter((location) => store.dispatch(navigate(location)));
+    window.history.pushState(null, document.title, '/new-route');
+    window.dispatchEvent(new PopStateEvent('popstate'));
```

Comment:
Isn't mimicking the browser the point here though?

We could trigger a popstate event by pushing twice then going back by 1, but that's really just a more convoluted way to do the exact same thing, whereas this is basically just manually triggering a click on a link.

Also, if you are using this in an app shell without redux, it may not be as easy to just call the callback itself.

Reply:
First, I think `new PopStateEvent('popstate')` will not work on IE so maybe need to change to something like `new CustomEvent('popstate')`. Second, I also think it's better to invoke the callback directly if possible to avoid side effect. So I'd prefer to have that as the recommended approach. We have similar suggestion in the [pwa-starter-kit wiki](https://github.com/Polymer/pwa-starter-kit/wiki/3.-Configuring-and-personalizing#routing)

Reply:
@jsilvermist This example is for going _forward_ in navigation without a link, so going forward then immediately going back (`popstate`) just to invoke the callback is convoluted. The code sample is to show that you should invoke whatever callback _you passed to `installRouter`_ - outside of Redux it wouldn't be `store.dispatch()`, hence the `// or whatever` comment.